src/Plugin/monitoring/SensorPlugin/TideSmtpSensorPlugin.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\tide_core\Plugin\monitoring\SensorPlugin;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\monitoring\Entity\SensorConfig;
+use Drupal\monitoring\Result\SensorResultInterface;
+use Drupal\monitoring\SensorPlugin\SensorPluginBase;
+use Drupal\monitoring\SensorPlugin\SensorPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Monitors the SMTP connection.
+ *
+ * @SensorPlugin(
+ *   id = "tide_smtp",
+ *   label = @Translation("Tide SMTP Sensor"),
+ *   description = @Translation("Monitors connectivity to SMTP"),
+ * )
+ */
+class TideSmtpSensorPlugin extends SensorPluginBase implements SensorPluginInterface {
+  
+  /**
+   * The module handler.
+   *
+   * @var Drupal\Core\Extension\ModuleHandler
+   */
+  protected $moduleHandler;
+  
+  public function __construct(SensorConfig $sensor_config, $plugin_id, $plugin_definition, ModuleHandlerInterface $module_handler) {
```

Comment:
Is this constructor necessary? It was in the sensor plugin I copied - but its not in the other one in this project.

Reply:
it looks necessary for dependency injection.

Reply:
or you can removed all of these `__construct`, `create` and `protected $moduleHandler`, and just use 
`\Drupal::service('module_handler')` installed.